src/Compilers/CSharp/Portable/Binder/WithLambdaParametersBinder.cs

```diff
@@ -29,24 +28,17 @@ public WithLambdaParametersBinder(LambdaSymbol lambdaSymbol, Binder enclosing)
             var parameters = lambdaSymbol.Parameters;
             if (!parameters.IsDefaultOrEmpty)
             {
-                recordDefinitions(parameters);
-                foreach (var parameter in lambdaSymbol.Parameters)
+                _definitionMap = new SmallDictionary<string, ParameterSymbol>();
+                foreach (var parameter in parameters)
                 {
                     if (!parameter.IsDiscard)
                     {
-                        this.parameterMap.Add(parameter.Name, parameter);
-                    }
-                }
-            }
-
-            void recordDefinitions(ImmutableArray<ParameterSymbol> definitions)
-            {
-                var declarationMap = _definitionMap ??= new SmallDictionary<string, ParameterSymbol>();
-                foreach (var s in definitions)
-                {
-                    if (!s.IsDiscard && !declarationMap.ContainsKey(s.Name))
-                    {
-                        declarationMap.Add(s.Name, s);
+                        var name = parameter.Name;
+                        this.parameterMap.Add(name, parameter);
+                        if (!_definitionMap.ContainsKey(name))
```

Comment:
We don't need the `IsDiscard` check after all (either on `parameterMap` or on `_definitionMap`)?

Reply:
Oops, missed it above ;-)